### PR TITLE
Get kth last element 15

### PR DESCRIPTION
**Pull request by samedkocyigit**

### CODE REVIEW

File: InterviewQuestions/Chapters/CustomLinkedLists/CustomLinkedList.cs [hunk 1]
```diff
@@ -1,39 +1,69 @@
...
+                while (currentNode.next != null)
+                {
+                    currentNode = currentNode.next;
+                }
+                currentNode.next = newNode;
+            }
+        }
+        public int getKhtLastElement(int index)
+        {
+            Node? currentNode = head;
```
Comment (furkankocyigit) [inline InterviewQuestions/Chapters/CustomLinkedLists/CustomLinkedList.cs:43]: Please separate the implementation of the length calculation to another method.

File: InterviewQuestions/Chapters/CustomLinkedLists/CustomLinkedList.cs [hunk 2]
```diff
@@ -1,39 +1,69 @@
...
+            int lenght = 0;
+            while (currentNode != null) 
+            {
+                currentNode = currentNode.next;
+                lenght++;
+            }
+            if (lenght < index)
+            {
+                Debug.Assert(false);
```
Comment (furkankocyigit) [inline InterviewQuestions/Chapters/CustomLinkedLists/CustomLinkedList.cs:52]: instead of debug.assert, throw an exception here. and also write unit tests for it. It would be good practice to check if your method throws an exception on the bad cases.

File: InterviewQuestionsTests/Chapters/CustomLinkedLists/CustomLinkedList.Tests.cs [hunk 3]
```diff
@@ -1,28 +1,40 @@
...
+            list.add(1);
+            list.add(2);
+            list.add(3);
+            list.add(4);
+            list.add(5);
+
+            Assert.AreEqual(list.getKhtLastElement(1), 5);
+            Assert.AreEqual(list.getKhtLastElement(3), 3);
+            Assert.AreEqual(list.getKhtLastElement(5), 1);
+        }
```
Comment (furkankocyigit) [inline InterviewQuestionsTests/Chapters/CustomLinkedLists/CustomLinkedList.Tests.cs:38]: check out of boundaries as well. method should not chrash

**Review by furkankocyigit — CHANGES_REQUESTED** [review]

File: InterviewQuestions/Chapters/CustomLinkedLists/CustomLinkedList.cs (hunk 2 above)
Comment (samedkocyigit) [reply InterviewQuestions/Chapters/CustomLinkedLists/CustomLinkedList.cs]: throw an exception instead of debug.assert. look here 9255e5871f9a8e4e3cd09e462367b94ad4ba77f4

File: InterviewQuestionsTests/Chapters/CustomLinkedLists/CustomLinkedList.Tests.cs (hunk 3 above)
Comment (samedkocyigit) [reply InterviewQuestionsTests/Chapters/CustomLinkedLists/CustomLinkedList.Tests.cs:38]: i do not get it exactly what to do?

File: InterviewQuestions/Chapters/CustomLinkedLists/CustomLinkedList.cs [hunk 4]
```diff
@@ -1,39 +1,73 @@
...
+            {
+                Node currentNode = head;
+                while (currentNode.next != null)
+                {
+                    currentNode = currentNode.next;
+                }
+                currentNode.next = newNode;
+            }
+        }
+        public int getLenghtLinkedList()
```
Comment (furkankocyigit) [inline InterviewQuestions/Chapters/CustomLinkedLists/CustomLinkedList.cs:41]: Just getLenght() would be better

File: InterviewQuestionsTests/Chapters/CustomLinkedLists/CustomLinkedList.Tests.cs (hunk 3 above)
Comment (furkankocyigit) [reply InterviewQuestionsTests/Chapters/CustomLinkedLists/CustomLinkedList.Tests.cs:38]: let's say you have a linked list whose length is 5.
what happens if I call list.getKhtLastElement(7) ??
so add a new unit test that checks the out of the boundaries as I said above, and be sure that expected exceptions are thrown.

**Hint**: search how to unit test expected exceptions in c#

**Review by furkankocyigit — CHANGES_REQUESTED** [review]

File: InterviewQuestions/Chapters/CustomLinkedLists/CustomLinkedList.cs (hunk 4 above)
Comment (samedkocyigit) [reply InterviewQuestions/Chapters/CustomLinkedLists/CustomLinkedList.cs]: Done. You can look here bc628d64f6458686147b1cd8f98a8020470fc47d

**Review by furkankocyigit — APPROVED** [review]
Thanks for clean solution. I have added my alternative solution, try to understand it, you can see it here -> b1b6862607ee254a67f24f59ebd1a7bd5b752834

And then feel free te merge this PR
